lib/auth.coffee

```diff
@@ -234,23 +238,102 @@ getAuth = (deps, opts) ->
 	# });
 	#
 	# @example
+	# resin.auth.createApiKey(apiKeyName, apiKeyDescription).then(function(apiKey) {
+	# 	console.log(apiKey);
+	# });
+	#
+	# @example
 	# resin.auth.createApiKey(apiKeyName, function(error, apiKey) {
 	# 	if (error) throw error;
 	# 	console.log(apiKey);
 	# });
 	###
-	exports.createApiKey = (name, callback) ->
+	exports.createApiKey = (name, description = null, callback) ->
+		callback = findCallback(arguments)
+
+		apiKeyBody =
+			name: name
+
+		if isString(description) and !!description
+			apiKeyBody.description = description
+
 		request.send
 			method: 'POST'
 			url: '/api-key/user/full'
 			baseUrl: apiUrl
-			body:
-				name: name
+			body: apiKeyBody
 		.get('body')
 		.catch ->
 			throw new errors.ResinNotLoggedIn()
 		.asCallback(callback)
 
+	###*
+	# @summary Get all API keys
+	# @name getAllApiKeys
+	# @public
+	# @function
+	# @memberof resin.auth
+	#
+	# @param {Object} [options={}] - extra pine options to use
+	# @fulfil {Object[]} - apiKeys
+	# @returns {Promise}
+	#
+	# @example
+	# resin.auth.getAllApiKeys().then(function(apiKeys) {
+	# 	console.log(apiKeys);
+	# });
+	#
+	# @example
+	# resin.auth.getAllApiKeys(function(error, apiKeys) {
+	# 	if (error) throw error;
+	# 	console.log(apiKeys);
+	# });
+	###
+	exports.getAllApiKeys = (options = {}, callback) ->
+		callback = findCallback(arguments)
+
+		return pine.get
+			resource: 'api_key'
+			options:
+				mergePineOptions
+					# the only way to reason whether
+					# it's a named user api key is whether
+					# it has a name
+					$filter: name: $ne: null
+					$orderby: 'name asc'
+				, options
+
+		.asCallback(callback)
+
+	###*
+	# @summary Revoke an API key
+	# @name revokeApiKey
+	# @public
+	# @function
+	# @memberof resin.auth
+	#
+	# @param {Number} id - API key id
+	# @returns {Promise}
+	#
+	# @example
+	# resin.auth.revokeApiKey(123);
+	#
+	# @example
+	# resin.auth.revokeApiKey(123, function(error) {
+	# 	if (error) throw error;
+	# });
+	###
+	exports.revokeApiKey = (id, callback) ->
```

Comment:
I ended up using the id here since the `name` field is not unique (natural key) for the api keys!
What I don't like at all is that `createApiKey` returns a string and need to call `getAllApiKeys` to pick the latest key by `created_at` (which arguably doesn't feel too robust given that `name`s are not unique).

Reply:
Ah, this is awkward. According to the spec, it looks like these should actually be unique: https://github.com/resin-io/hq/pull/861/files#diff-18d34b62d6da8f468178ae8c7359b74fR44

> We won't allow users to repeat names across their API keys, they will act as human-readable identifier within each account

It does seem that you're correct and we haven't enforced that though.

For now let's use an id here anyway. Later, I think we should make names unique (checking with Shaun and somehow migrating in that direction), and then later we can make this method take both strings or numbers, and revoke the name/id accordingly.